torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -801,6 +809,11 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // Returns the global ranks of a PG.
   const std::vector<uint64_t>& groupRanks() const;
 
+  // Util function to assign timeout to each work.
+  void assignTimeoutToWork(
+      c10::intrusive_ptr<ProcessGroupNCCL::WorkNCCL> work,
+      c10::intrusive_ptr<Options> option);
```

Comment:
Options 🤔 

Are you sure you want to take ownership for these? Seems unlikely...

Reply:
They are passed in from ctor. We don't take ownership of Options.

Reply:
If you don't take ownership, pass by const ref

Reply:
Sure.